apps/dashboard/app/models/configuration_property.rb

```diff
@@ -0,0 +1,61 @@
+class ConfigurationProperty
+
+  def self.with_boolean_mapper(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, BooleanMapper)
+  end
+
+  def self.with_integer_mapper(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, IntegerMapper)
+  end
+
+  def self.property(name:, default_value: nil, read_from_env: false, env_name: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_name, PassThroughMapper)
+  end
+
+  attr_reader :name, :default_value, :read_from_environment, :environment_name
+  alias_method :read_from_environment?, :read_from_environment
+
+  def initialize(name, default_value, read_from_env, env_name, mapper)
+    @name = name.to_sym
+    @default_value = default_value
+    @read_from_environment = read_from_env
+    @mapper = mapper
+
+    environment_name = env_name || "OOD_#{@name.to_s.upcase}"
+    @environment_name = environment_name if @read_from_environment
+  end
+
+  def map_string(value)
+    @mapper.map_string(value)
+  end
```

Comment:
It seems like this shouldn't need the input.

From reading the usage in `user_configuration` it seems like this is actually `to_s` which shouldn't require an input argument and a you'd need an `from_env` API. 

```suggestion
  def to_s
    @mapper.map_string(@some_internal_value)
  end
```

We can defer this and refactor a bit later, I just wonder if we can actually encapsulate this a little better.

Reply:
Initially I thought about having the environment lookup in `ConfigurationProperty`, but then I wanted all lookup logic to be in UserConfiguration instead of across 2 classes.

I am fine either way, but I think the name of the method should be more significant: `from_env`, `read_from_env`, `env_value`

Reply:
Yea I've got a ticket to defer it. It's one of those things you'd have to see both to make a decision anyhow, so we'll leave it.